tests/syntax/test_requirements.py

```diff
@@ -315,3 +315,42 @@ def test_static_intersection_violation_disabled():
         ego = new Object at 0@0
         new Object at 1@0, with allowCollisions True
     """)
+
+## Tests for random properties that have special effects on constructing requirements
+def test_random_allowCollisions():
+    for _ in range(5):
+        sampleSceneFrom("""
+            new Object with allowCollisions Uniform(True, False)
+            new Object with allowCollisions Uniform(True, False)
+        """, maxIterations=10)
```

Comment:
Each iteration of the loop has about a 1 in 10^6 chance of failing, which is too high (this test may get run many times in CI, etc.). I'd increase `maxIterations` to 30. This doesn't slow the test down in any case.

I'm not sure what the outer loop is for: if the point is to ensure we get at least one round where more than 1 iteration is necessary, we'd need more repetitions to have a good probability of that. (But you could break out of the loop as soon as one of the calls uses more than 1 iteration.)